numpy/core/src/multiarray/common_dtype.c

```diff
@@ -35,13 +72,19 @@
  *
  * TODO: Before exposure, we should review the return value (e.g. no error
  *       when no common DType is found).
+ *       Further, the `object_fallback` is probably only useful for the
```

Comment:
If you are fairly sure that in some places this will no longer be needed after the string deprecation, maybe be more explicit about it, otherwise in all likelihood this will stay around for much longer than needed!